### DISCUSSION TITLE
The findAll/findAndCount method returns only one record v6.2.x -> v6.3.13

### DISCUSSION THREAD

**Discussion by mikara89** [mikro-orm/mikro-orm#6166] [General]
### Describe the bug

The findAll/findAndCount method returns only one record v6.2.x -> v6.3.13. version 6.1.0 works fine.

### Reproduction

```   
const queryResult = await this.profileRepository
      .findAndCount(
        null,
        {
          limit: 10,
          offset: 0,
          disableIdentityMap: true,
          flags: [QueryFlag.PAGINATE],
        },
      );
```
Result:
```
0 = (1) [Profile] //expect to get 10 of total 11
1 = 11
length =2
[[Prototype]] =Array(0)
[[Prototype]] =Object
```
sql:
```
select "p0".* 
from "profile" as "p0" 
where "p0"."id" in 
  (select "p0"."id" from 
      (select "p0"."id" from "profile" as "p0" group by "p0"."id" limit 10) 
  as "p0")
---------
"id"	"name"	"created_at"	"updated_at"	"created_by"
1	"string"	"2024-10-19 19:32:01.324+00"	"2024-10-19 19:32:01.324+00"	"00000000-0000-0000-0000-000000000000"
2	"string2"	"2024-10-19 19:37:55.431+00"	"2024-10-19 19:37:55.432+00"	"00000000-0000-0000-0000-000000000000"
3	"string1"	"2024-10-19 19:38:35.562+00"	"2024-10-19 19:38:35.562+00"	"00000000-0000-0000-0000-000000000000"
4	"string3"	"2024-10-19 19:39:51.297+00"	"2024-10-19 19:39:51.297+00"	"00000000-0000-0000-0000-000000000000"
5	"string4"	"2024-10-19 19:40:45.022+00"	"2024-10-19 19:40:45.022+00"	"00000000-0000-0000-0000-000000000000"
6	"string5"	"2024-10-19 19:43:42.131+00"	"2024-10-19 19:43:42.131+00"	"00000000-0000-0000-0000-000000000000"
7	"string6"	"2024-10-19 19:44:24.243+00"	"2024-10-19 19:44:24.243+00"	"00000000-0000-0000-0000-000000000000"
8	"string7"	"2024-10-19 19:46:23.14+00"	"2024-10-19 19:46:23.14+00"	"00000000-0000-0000-0000-000000000000"
9	"string8"	"2024-10-19 19:52:12.913+00"	"2024-10-19 19:52:12.913+00"	"00000000-0000-0000-0000-000000000000"
11	"string9"	"2024-10-19 20:08:37.962+00"	"2024-10-19 20:08:37.962+00"	"00000000-0000-0000-0000-000000000000"
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.x -> 6.3.13

### Node.js version

20.14.0

### Operating system

Win11

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
This is not a [reproduction](https://github.com/mikro-orm/reproduction), I can't help without seeing what you are actually doing. Note that you are not supposed to be passing `null` to the where condition, that code won't even compile.

**Reply by mikara89** [reply]
Thanks for taking into consideration my problem, it is not easy to reproduce but here you go:
[reproduction ](https://github.com/mikara89/reproduction )
this is verson 6.3.13 and if you downgrade to 6.1.0 works fine

**Reply by B4nan** [reply]
The problem here is that your custom ID class is not serializable. Add `toString` method to it and it will work correctly:

```ts
export class UserId {

  // ...

  toString() {
    return this._value;
  }

}
```

**Reply by mikara89** [reply]
Ok, thanks, it will be usefull to have warning on that part, if not error :) Many thanks! PS: v6.1.5 is last woking version if I dont have `toString()`

**Reply by B4nan** [reply]
Indeed, this is just a workaround. Here is a proper fix: aa1dd3c101f187df5ca425c4b0468f65df47ffaa
